x/incentives/keeper/gauge.go

```diff
@@ -110,13 +111,41 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		}
 	}
 
-	// check if denom this gauge pays out to exists on-chain
-	if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
-		return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
+	nextGaugeId := k.GetLastGaugeID(ctx) + 1
+
+	// For no lock gauges, a pool id must be set.
+	// A pool with such id must exist and be a concentrated pool.
+	if distrTo.LockQueryType == lockuptypes.NoLock {
+		if poolId == 0 {
+			return 0, fmt.Errorf("no lock gauges must have a pool id")
+		}
+
+		pool, err := k.pmk.GetPool(ctx, poolId)
+		if err != nil {
+			return 0, err
+		}
+
+		if pool.GetType() != poolmanagertypes.Concentrated {
+			return 0, fmt.Errorf("no lock gauges must be created for concentrated pools only")
+		}
+
+		// We assume that external gauges are created with 0 duration,
+		// while internal gauges are created with an incentive epoch duration.
+		k.pik.SetPoolGaugeId(ctx, poolId, distrTo.Duration, nextGaugeId)
```

Comment:
Nice catch, the query needs to be fixed to avoid excluding "NoLock" type gauges that have a denom made from `incentivestyps.NoLockInternalGaugeDenom(<pool id>)`

Reply:
I think I need to also fix the `IncentivizedPools` query. I will fix and test both

Reply:
wait are you sure it's fixed? 
`QueryExternalIncentiveGaugesRequest` iterates on "pool-incentives" prefix and gets gauges without the prefix.

in this case the external gauges will contain "pool-incentives" prefix no?

Reply:
also CLI tested this and it doesnot seem to be working as expected

Reply:
`QueryExternalIncentiveGaugesRequest` definitely needs fixin and tests. `IncentivizedPools` is likely fine but I will add more tests just in case.

I will ping you once R4R for re-review

Reply:
sounds good, meantime i'll check `IncentivizedPools`

Reply:
Fix: https://github.com/osmosis-labs/osmosis/pull/5475

Please see PR description for details

Reply:
Merged the fix from the above PR into this branch in: [c9760e6](https://github.com/osmosis-labs/osmosis/pull/5459/commits/c9760e656676090942fc08dabb8503e65826e142)

Going to resolve now